pkg/shared/utils/sync/distributed_lock.go

```diff
@@ -0,0 +1,54 @@
+package sync
+
+import (
+	"github.com/pkg/errors"
+	"gorm.io/gorm"
+)
+
+// DistributedLock manages locks on the database through the usage of the `distributed_lock` table.
+// WARNING: a lock can be released only by the same instance that created it.
+type DistributedLock interface {
+	Lock() error
+	Unlock() error
+}
```

Comment:
Can we extend of this PR onto using the lock over here https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/ed0c44234c8ff98dbe2be35e3bd42b5b52aa69bc/internal/kafka/internal/services/kafkatlscertmgmt/secure_storage.go#L66-L93 ? Thanks

Reply:
The integration could be a separate commit if we prefer

Reply:
Done!

Reply:
Thanks, current checking it out